### ISSUE TITLE
关于A-softmax的知识点问题

### ISSUE THREAD

**Issue by woshildh** [wujiyang/Face_Pytorch#21]
您好，我正在尝试复现A-softmax。这是您的代码中的相关操作，想问一下您这个cur_lambda和Iter的作用是什么？非常感谢
`
def forward(self, input, label):
        self.iter += 1
        self.cur_lambda = max(self.lambda_min, self.base * (1 + self.gamma * self.iter) ** (-1 * self.power))

        cos_theta = F.linear(F.normalize(input), F.normalize(self.weight))
        cos_theta = cos_theta(-1, 1)

        cos_m_theta = self.margin_formula(self.m)(cos_theta)
        theta = cos_theta.data.acos()
        k = ((self.m * theta) / math.pi).floor()
        phi_theta = ((-1.0) ** k) * cos_m_theta - 2 * k
        phi_theta_ = (self.cur_lambda * cos_theta + phi_theta) / (1 + self.cur_lambda)
        norm_of_feature = torch.norm(input, 2, 1)

        one_hot = torch.zeros_like(cos_theta)
        one_hot.scatter_(1, label.view(-1, 1), 1)

        output = one_hot * phi_theta_ + (1 - one_hot) * cos_theta
        output *= norm_of_feature.view(-1, 1)

        return output
`

**Comment by wujiyang** [comment]
A-softmax在实际实现的时候，并不仅仅使用了 multiplicative magin，他在计算loss的时候使用的是A-softmax和softmax的均值，即（A-softmax + lambda * softmax）/ ( 1 + lambda)

**Comment by woshildh** [comment]
好的，非常感谢 ☺
